gcc/rust/backend/rust-compile-expr.cc

```diff
@@ -223,6 +224,13 @@ CompileExpr::visit (HIR::MatchExpr &expr)
       rust_assert (adt->is_enum ());
       rust_assert (adt->number_of_variants () > 0);
     }
+  else if (scrutinee_kind == TyTy::TypeKind::FLOAT)
+    {
+      // FIXME: CASE_LABEL_EXPR does not support floating point types.
+      // Find another way to compile these.
+      sorry_at (expr.get_locus ().gcc_location (),
```

Comment:
This is great :D I think it would be nice to have a `rust_sorry` wrapper to which we can give a `Location` directly instead of a `location_t`. But in another PR :)